server/types/record.go

```diff
@@ -51,5 +55,11 @@ var Record = &DoltgresType{
 	Acl:           nil,
 	Checks:        nil,
 	attTypMod:     -1,
-	CompareFunc:   toFuncID("btrecordcmp", toInternal("record"), toInternal("record")),
+}
+
+// RecordValue holds the value of a single field in a record, including type information for the
+// field value.
+type RecordValue struct {
+	Value any
+	Type  sql.Type
```

Comment:
Any reason this is `sql.Type` instead of our Doltgres type?

Reply:
Hrm... I was originally seeing some `sql.NullType` values, so kept this as `sql.Type`, but I just tested and don't see that happening anymore. I changed this to `DoltgresType` and it cleaned up some of the other type checks elsewhere in the code.